diff --git a/node_modules/file-selector/dist/es2015/file-selector.js b/node_modules/file-selector/dist/es2015/file-selector.js
index 6f318e7..2e2acfc 100644
--- a/node_modules/file-selector/dist/es2015/file-selector.js
+++ b/node_modules/file-selector/dist/es2015/file-selector.js
@@ -111,6 +111,7 @@ function fromDataTransferItem(item, entry) {
     if (typeof item.getAsFileSystemHandle === 'function') {
         return item.getAsFileSystemHandle()
             .then((h) => __awaiter(this, void 0, void 0, function* () {
+            console.log("h is", h)
             const file = yield h.getFile();
             file.handle = h;
             return toFileWithPath(file);
diff --git a/node_modules/file-selector/src/file-selector.ts b/node_modules/file-selector/src/file-selector.ts
index 4aaee19..95ecd54 100644
--- a/node_modules/file-selector/src/file-selector.ts
+++ b/node_modules/file-selector/src/file-selector.ts
@@ -23,7 +23,7 @@ export async function fromEvent(evt: Event | any): Promise<(FileWithPath | DataT
         return getDataTransferFiles(evt.dataTransfer, evt.type);
     } else if (isChangeEvt(evt)) {
         return getInputFiles(evt);
-    } else if (Array.isArray(evt) && evt.every(item => 'getFile' in item && typeof item.getFile === 'function')) {
+    } else if (Array.isArray(evt) && evt.every(item => 'getFile' in item && typeof item?.getFile === 'function')) {
         return getFsHandleFiles(evt)
     }
     return [];
@@ -47,7 +47,7 @@ function getInputFiles(evt: Event) {
 
 // Ee expect each handle to be https://developer.mozilla.org/en-US/docs/Web/API/FileSystemFileHandle
 async function getFsHandleFiles(handles: any[]) {
-    const files = await Promise.all(handles.map(h => h.getFile()));
+    const files = await Promise.all(handles.map(h => h?.getFile()));
     return files.map(file => toFileWithPath(file));
 }
 
@@ -124,7 +124,8 @@ function fromDataTransferItem(item: DataTransferItem, entry?: FileSystemEntry |
     if (typeof (item as any).getAsFileSystemHandle === 'function') {
         return (item as any).getAsFileSystemHandle()
             .then(async (h: any) => {
-                const file = await h.getFile();
+                console.log("h is", h)
+                const file = await h?.getFile();
                 file.handle = h;
                 return toFileWithPath(file);
             });
